internal/oc/release.go

```diff
@@ -126,7 +127,16 @@ func (r *release) GetMustGatherImage(log logrus.FieldLogger, releaseImage string
 
 // GetCoreOSImage gets rhel-coreos image URL from the release image or releaseImageMirror, if provided.
 func (r *release) GetCoreOSImage(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error) {
-	return r.getImageByName(log, coreosImageName, releaseImage, releaseImageMirror, pullSecret)
+	imageName := coreosImageName
+	if isOKD(releaseImage) {
+		imageName = streamCoreOSImageName
+	}
+	return r.getImageByName(log, imageName, releaseImage, releaseImageMirror, pullSecret)
+}
+
+// Returns true if the image matches OKD release image pattern
+func isOKD(image string) bool {
```

Comment:
Is this consistent with how we're checking for OKD in other places? I remember seeing a check somewhere but I don't know how it's implemented.

Reply:
I actually think that is how we use that function. The problem is that we end up running it all the time which isn't great but ... I suppose it depends on how confident we are in the release image naming.

Reply:
@Prashanth684 would you have any recommendation on how to detect reliably OKD release image?

Reply:
unfortunately, there is no easy way to detect if  release image is OKD outside of doing what oc does which is read the release image, parse the metadata and maybe check the componentversion to see if there is centos stream coreos string present there. The only way today is like you suggest to check for both and see which one works.